### PR TITLE
add ver argument for init in subclasses of ValidHDU.

### PR DESCRIPTION
**Pull request by MSeifert04**
This is WIP for now (passes locally but I want to see if it passes against the complete matrix). 

I'm still testing why it's not working if I pass the `**kwargs` to `super` in `ImageHDU.__init__`.

Related #6453

### CODE REVIEW

**Comment by pllim** [comment]
This is probably not a bug fix, strictly speaking, so marking for v3 for now. Feel free to re-milestone if you want.

**Comment by MSeifert04** [comment]
@pllim I have no editing rights for milestones and labels.

But could you stop the tests? I just figured out that I haven't commited the latest commit. 😅

**Comment by pllim** [comment]
I cancelled Appveyor and Travis but looks like it just picked up your second commit again (squash + force push?), so I think everything is in order? I don't have access to CircleCI.

**Comment by MSeifert04** [comment]
@pllim Yes, it seems like it picked up the squashed commits. It just took a few minutes (no idea why, maybe just a github hickup)...

**Comment by MSeifert04** [comment]
ok, I don't think I'll figure out how to properly pass the arguments up to the superclasses so that it works correctly.

**Comment by MSeifert04** [comment]
Yeah, I found out why it's not a good idea to "pass name up" the inheritance hierarchy. It's because `PrimaryHDU`s also subclass `_ImageHDU`. From a casual overview it would take some refactoring (I made it pass *almost* all the test cases) to make it work correctly and then who knows what "user-code" that might break.

The current changes aren't very big and it works. Maybe better to just leave it at that?!

cc @saimn @pllim

File: astropy/io/fits/hdu/image.py [hunk 1]
```diff
@@ -1087,6 +1090,9 @@ def __init__(self, data=None, header=None, name=None,
             operations on the data.  Pseudo-unsigned integers are automatically
             rescaled unless scale_back is explicitly set to `False`.
             (default: None)
+
+        ver : int > 0, optional
+            The name of the HDU, will be the value of the keyword ``EXTVER``.
```
Comment (pllim) [inline astropy/io/fits/hdu/image.py]: Maybe add a note that, if not given, it will be set to 1 even in the case of duplicate `EXTNAME` in a HDU list?

Comment (dborncamp) [inline astropy/io/fits/hdu/image.py]: That will work to at least make a fits compliant file. Thanks @pllim and @MSeifert04 !

**Comment by pllim** [comment]
I'll wait for @saimn to comment/approve before merge.

File: astropy/io/fits/hdu/image.py (hunk 1 above)
Comment (MSeifert04) [inline astropy/io/fits/hdu/image.py]: @pllim I clarified the docstring. It only defaults to `1` if `ver=None` and when the `header` has no `EXTVER`.

**Comment by MSeifert04** [comment]
Sorry, I must have forgotten but what does the `whatsnew-needed` label mean?

**Comment by pllim** [comment]
The label is a reminder to add an entry into "What's New" for tagged release. Or so I thought. I am not sure if this really warrants an entry there though it seems useful enough that we should do it. Maybe @bsipocz can clarify.

File: astropy/io/fits/hdu/image.py (hunk 1 above)
Comment (pllim) [inline astropy/io/fits/hdu/image.py]: I guess a test doesn't hurt? :smile:

File: astropy/io/fits/hdu/image.py [hunk 2]
```diff
@@ -1087,6 +1090,11 @@ def __init__(self, data=None, header=None, name=None,
             operations on the data.  Pseudo-unsigned integers are automatically
             rescaled unless scale_back is explicitly set to `False`.
             (default: None)
+
+        ver : int > 0 or None, optional
+            The ver of the HDU, will be the value of the keyword ``EXTVER``.
+            If not given or None it default to ``EXTVER`` or ``1``.
```
Comment (pllim) [inline astropy/io/fits/hdu/image.py]: Grammatical nitpicks:
* Comma after "None"
* default -> defaults

Comment (pllim) [inline astropy/io/fits/hdu/image.py]: Does 1 really need to be in double-backticks?

**Comment by MSeifert04** [comment]
I'm pretty certain I'm not responsible for this exception but it would be interesting why it's happening:

```
Exception occurred:

  File "/home/travis/build/astropy/astropy/build/lib.linux-x86_64-3.6/astropy/io/misc/connect.py", line 8, in <module>

    from .hdf5 import read_table_hdf5, write_table_hdf5, is_hdf5

ImportError: cannot import name 'read_table_hdf5'
The full traceback has been saved in /tmp/sphinx-err-5p122t8p.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
The build_docs travis build FAILED because sphinx issued documentation warnings (scroll up to see the warnings).
Sphinx Documentation subprocess failed with return code 1
```

Should I restart the test?

**Comment by pllim** [comment]
I restarted it, it got time out errors, so I restarted it again...

**Comment by bsipocz** [comment]
@pllim @MSeifert04 - In my understanding we keep the what's new mostly for big/significant changes and additions to highlight. I think a changelog entry for this PR should be enough, but @astrofrog or @eteq may comment on it.

**Comment by saimn** [comment]
Also I agree with @bsipocz, I don't think this deserves an entry in the *What's new* page.

**Comment by saimn** [comment]
Thanks for the quick update @MSeifert04, tests are passing so merging!

**Comment by pllim** [comment]
I'll spare you a squash. 😉  Thanks!

**Pull request merged**